sound/soc/soc-pcm.c

```diff
@@ -745,6 +749,9 @@ static int soc_pcm_close(struct snd_pcm_substream *substream)
 	for_each_rtdcom(rtd, rtdcom) {
 		component = rtdcom->component;
 
+		if (component->driver->module_get_upon_open)
+			module_put(component->dev->driver->owner);
```

Comment:
this isn't balanced with the module_get(), which is done only when you have a .ops and a .open. You might decrease references that were not previously incremented.

Reply:
@plbossart good catch. I did infact put it in the wrong place.